docs(fix): clarify how worker.version and minWorkerVersion comparison works (#12459)
apache/druid: docs/configuration/index.md

|Property|Description|Default|
|--------|-----------|-------|
|`druid.indexer.runner.taskAssignmentTimeout`|How long to wait after a task as been assigned to a MiddleManager before throwing an error.|PT5M|

> |`druid.indexer.runner.minWorkerVersion`|The minimum MiddleManager version to send tasks to. Note that the version number is a string. This affects the expected behavior during certain operations like when it's compared against `druid.worker.version`. Specifically, the version comparison follows dictionary order . Use ISO8601 date format for the version so that later versions are recognized as being more recent. |"0"|

Reviewer: ```suggestion
|`druid.indexer.runner.minWorkerVersion`|The minimum MiddleManager version to send tasks to. The version number is a string. This affects the expected behavior during certain operations like comparison against `druid.worker.version`. Specifically, the version comparison follows dictionary order. Use ISO8601 date format for the version to accommodate date comparisons. |"0"|
```

|`druid.indexer.runner.endPort`|Ending port used for peon processes, should be greater than or equal to `druid.indexer.runner.startPort` and less than 65536.|65535|
|`druid.indexer.runner.ports`|A JSON array of integers to specify ports that used for peon processes. If provided and non-empty, ports for peon processes will be chosen from these ports. And `druid.indexer.runner.startPort/druid.indexer.runner.endPort` will be completely ignored.|`[]`|
|`druid.worker.ip`|The IP of the worker.|localhost|

> |`druid.worker.version`|Version identifier for the MiddleManager. The version number is a string. This affects the expected behavior during certain operations like comparison against `druid.indexer.runner.minWorkerVersion`. Specifically, the version comparison follows dictionary order. Use ISO8601 date format for the version to accomodate date comparisons.|0|

Reviewer: ```suggestion
|`druid.worker.version`|Version identifier for the MiddleManager. The version number is a string. This affects the expected behavior during certain operations like comparison against `druid.indexer.runner.minWorkerVersion`. Specifically, the version comparison follows dictionary order. Use ISO8601 date format for the version to accommodate date comparisons.|0|
```

> |`druid.worker.version`|Version identifier for the MiddleManager. Note that the version number is a string. This affects the expected behavior during certain operations like when it's compared against `druid.indexer.runner.minWorkerVersion`. Specifically, the version comparison follows dictionary order. Use ISO8601 date format for the version so that later versions are recognized as being more recent.|0|

Reviewer: see line 1081

> |`druid.worker.version`|Version identifier for the MiddleManager. Note that the version number is a string. This affects the expected behavior during certain operations like when it's compared against `druid.indexer.runner.minWorkerVersion`. Specifically, the version comparison follows dictionary order. Use ISO8601 date format for the version so that later versions are recognized as being more recent.|0|

Author: ```suggestion
|`druid.worker.version`|Version identifier for the MiddleManager. The version number is a string. This affects the expected behavior during certain operations like comparison against `druid.indexer.runner.minWorkerVersion`. Specifically, the version comparison follows dictionary order. Use ISO8601 date format for the version to accomodate data comparisons.|0|
```
